test/test_results.py

```diff
@@ -0,0 +1,158 @@
+import unittest
+import sys
+import os
+import subprocess
+import warnings
+
+# Add the parent directory to the Python path
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from test.utils.custom_tsv_parser import ParseTSVFile
+
+PREF_CUTOFF = -4
+
+
+def get_scode_cscode_id(data):
+    # Create an id with scode and center with sorted values of scode and center
+    school_centers = []
+    for row in data:
+        scode = int(row["scode"])
+        center = int(row["cscode"])
+        sccode_center_id = sorted((scode, center))
+        sccode_center_id = "_".join(map(str, sccode_center_id))
+        school_centers.append(sccode_center_id)
+    return school_centers
+
+
+class TestSchoolCenter(unittest.TestCase):
+    """_Tests to validate the outcome of the output are matching
+    as per the requirements
+
+    Needs the ouptut from the results/school-center.tsv and
+    results/school-center-distance.tsv files to be present
+
+    """
+
+    def setUp(self):
+        self.school_center_file = "results/school-center.tsv"
+        self.school_center_distance_file = "results/school-center-distance.tsv"
+        self.school_center_pref_file = "sample_data/prefs.tsv"
+        schools_tsv = "sample_data/schools_grade12_2081.tsv"
+        centers_tsv = "sample_data/centers_grade12_2081.tsv"
+        cmd = f"python school_center.py {schools_tsv} {centers_tsv} {self.school_center_pref_file}"
+        subprocess.run(cmd, shell=True)
+
+    def tearDown(self):
+        os.remove(self.school_center_file)
+        os.remove(self.school_center_distance_file)
+
+    def test_results_exists(self):
+        """_Test if the application in running which output the results in the
+             results filder_
+
+        Returns:
+            Pass: If the file exists in the results folder
+            Fail: If the file doesnot exists in the results folder
+        """
+        self.assertTrue(os.path.exists(self.school_center_file))
+        self.assertTrue(os.path.exists(self.school_center_distance_file))
+        self.assertTrue(os.path.exists(self.school_center_pref_file))
+
+    def test_scode_student_count_not_more_than_200(self):
+        """_Test if the student count is not more than 200_
+          Test case ID 001:- एक विद्यालयको परिक्षार्थी संख्या हेरी सकभर १००, २०० भन्दा बढी
+                             परीक्षार्थी एकै केन्द्रमा नपर्ने गरी बाँढ्न पर्ने
+
+        Returns:
+            Pass: If the student count is not more than 100
+            Fail: If the student count is more than 100
+        """
+        ptf = ParseTSVFile(self.school_center_file)
+        data = ptf.get_rows()
+        for row in data:
+            student_count = row["allocation"]
+            if int(student_count) > 200:
+                warnings.warn(f"student count is more than 200 for the school {row}")
```

Comment:
Why are we logging here instead of failing ?

Reply:
As per PR review previously. The 200 limit is not a hard limit and was advised to not fail the test so warning is the best bet at least gives awareness to the tester.